src/main/java/com/minecolonies/apiimp/initializer/ModBuildingsInitializer.java

```diff
@@ -606,7 +607,28 @@ public static void init(final RegistryEvent.Register<BuildingEntry> event)
                                     .setBuildingViewProducer(() -> EmptyView::new)
                                     .setRegistryName(new ResourceLocation(Constants.MOD_ID, ModBuildings.GRAVEYARD_ID))
                                     .addBuildingModuleProducer(GraveyardManagementModule::new, () -> GraveyardManagementModuleView::new)
-                                   .addBuildingModuleProducer(() -> new WorkerBuildingModule(ModJobs.undertaker, Skill.Strength, Skill.Mana, false, (b) -> 1), () -> WorkerBuildingModuleView::new)
+                                    .addBuildingModuleProducer(() -> new WorkerBuildingModule(ModJobs.undertaker, Skill.Strength, Skill.Mana, false, (b) -> 1), () -> WorkerBuildingModuleView::new)
+                                    .createBuildingEntry();
+
+        ModBuildings.simpleQuarry = new BuildingEntry.Builder()
+                                   .setBuildingBlock(ModBlocks.blockSimpleQuarry)
+                                   .setBuildingProducer((colony, blockPos) -> new DefaultBuildingInstance(colony, blockPos, "simplequarry", 1)).setBuildingViewProducer(() -> EmptyView::new)
+                                   .addBuildingModuleProducer(QuarryModule::new, () -> MinerAssignmentModuleView::new)
+                                   .setRegistryName(new ResourceLocation(Constants.MOD_ID, ModBuildings.SIMPLE_QUARRY_ID))
+                                   .createBuildingEntry();
+
+        ModBuildings.mediumQuarry = new BuildingEntry.Builder()
+                                   .setBuildingBlock(ModBlocks.blockMediumQuarry)
+                                   .setBuildingProducer((colony, blockPos) -> new DefaultBuildingInstance(colony, blockPos, "mediumquarry", 1)).setBuildingViewProducer(() -> EmptyView::new)
+                                   .addBuildingModuleProducer(QuarryModule::new, () -> MinerAssignmentModuleView::new)
+                                   .setRegistryName(new ResourceLocation(Constants.MOD_ID, ModBuildings.MEDIUM_QUARRY_ID))
+                                   .createBuildingEntry();
+
+        ModBuildings.largeQuarry = new BuildingEntry.Builder()
+                                   .setBuildingBlock(ModBlocks.blockLargeQuarry)
+                                   .setBuildingProducer((colony, blockPos) -> new DefaultBuildingInstance(colony, blockPos, "largequarry", 1)).setBuildingViewProducer(() -> EmptyView::new)
+                                   .addBuildingModuleProducer(QuarryModule::new, () -> MinerAssignmentModuleView::new)
+                                   .setRegistryName(new ResourceLocation(Constants.MOD_ID, ModBuildings.LARGE_QUARRY_ID))
```

Comment:
If the quarrier is another structure-builder type, they should probably get a `SimpleCraftingModule` so they can be taught to make torches at least.

Reply:
The quarrier is primarily deployed at the mine, so that will work directly.